packages/redux/src/products/reducer/__tests__/outfits.test.ts

```diff
@@ -0,0 +1,214 @@
+import { mockProductId } from 'tests/__fixtures__/products';
+import { productsActionTypes, ProductsOutfitsState } from '../..';
+import { toBlackoutError } from '@farfetch/blackout-client';
+import reducer, { getError, getIsLoading, INITIAL_STATE } from '../outfits';
+
+const mockAction = { type: 'foo' };
+const meta = { productId: mockProductId };
+let initialState: ProductsOutfitsState;
+
+describe('outfits redux reducer', () => {
+  beforeEach(() => {
+    initialState = reducer(INITIAL_STATE, mockAction);
+  });
+
+  describe('error() reducer', () => {
+    const error = new Error('An error occurred');
+    const expectedError = {
+      [mockProductId]: error,
+    };
+
+    it('should return the initial state', () => {
+      const state = reducer(INITIAL_STATE, mockAction).error;
+
+      expect(state).toEqual(initialState.error);
+    });
+
+    it('should handle FETCH_PRODUCT_OUTFITS_REQUEST action type with state state', () => {
+      const expectedResult = { [mockProductId]: undefined };
+      const state = reducer(undefined, {
+        meta,
+        type: productsActionTypes.FETCH_PRODUCT_OUTFITS_REQUEST,
+      });
+
+      expect(state.error).toEqual(expectedResult);
+    });
+
+    it('should handle FETCH_PRODUCT_OUTFITS_REQUEST action type with state', () => {
+      const expectedResult = {
+        888: toBlackoutError(error),
+        [mockProductId]: undefined,
+      };
+      const state = reducer(
+        {
+          error: { 888: toBlackoutError(error) },
```

Comment:
This test should contain an error entry for the same productId that is being dispatched (`mockProductId` in this case) so it can test that the previous state for that product id will be cleared.

Reply:
Yes, makes sense to me too.
Changed